src/bindings/c/ov/src/ov_c_api.cpp

```diff
@@ -0,0 +1,1310 @@
+}
+
+ov_status_e ov_infer_request_set_tensor(ov_infer_request_t *infer_request,
+                                const char* tensor_name, const ov_tensor_t *tensor) {
+    if (!infer_request || !tensor_name || !tensor) {
+        return ov_status_e::GENERAL_ERROR;
+    }
+
+    try {
+        infer_request->object->set_tensor(tensor_name, *tensor->object);
+    } CATCH_OV_EXCEPTIONS
+
+    return ov_status_e::OK;
+}
+
+ov_status_e ov_infer_request_set_input_tensor(ov_infer_request_t* infer_request,
+                                size_t idx, const ov_tensor_t* tensor) {
+    if (!infer_request || !tensor) {
+        return ov_status_e::GENERAL_ERROR;
+    }
+
+    try {
+        infer_request->object->set_input_tensor(idx, *tensor->object);
+    } CATCH_OV_EXCEPTIONS
+
+    return ov_status_e::OK;
+}
+
+ov_status_e ov_infer_request_get_tensor(const ov_infer_request_t* infer_request,
+                                const char* tensor_name, ov_tensor_t **tensor) {
+    if (!infer_request || !tensor_name || !tensor) {
+        return ov_status_e::GENERAL_ERROR;
+    }
+
+    try {
+        *tensor = new ov_tensor_t;
+        ov::Tensor tensor_get = infer_request->object->get_tensor(tensor_name);
+        (*tensor)->object = std::make_shared<ov::Tensor>(std::move(tensor_get));
+    } CATCH_OV_EXCEPTIONS
+
+    return ov_status_e::OK;
+}
+
+ov_status_e ov_infer_request_get_out_tensor(const ov_infer_request_t* infer_request,
+                                size_t idx, ov_tensor_t **tensor) {
+    if (!infer_request || !tensor) {
+        return ov_status_e::GENERAL_ERROR;
+    }
+
+    try {
+        *tensor = new ov_tensor_t;
+        ov::Tensor tensor_get = infer_request->object->get_output_tensor(idx);
+        (*tensor)->object = std::make_shared<ov::Tensor>(std::move(tensor_get));
+    } CATCH_OV_EXCEPTIONS
+
+    return ov_status_e::OK;
+}
+
+ov_status_e ov_infer_request_infer(ov_infer_request_t* infer_request) {
+    if (!infer_request) {
+        return ov_status_e::GENERAL_ERROR;
+    }
+
+    try {
+        infer_request->object->infer();
+    } CATCH_OV_EXCEPTIONS
+
+    return ov_status_e::OK;
+}
+
+ov_status_e ov_infer_request_start_async(ov_infer_request_t* infer_request) {
+    if (!infer_request) {
+        return ov_status_e::GENERAL_ERROR;
+    }
+
+    try {
+        infer_request->object->start_async();
+    } CATCH_OV_EXCEPTIONS
+
+    return ov_status_e::OK;
+}
+
+ov_status_e ov_infer_request_wait(ov_infer_request_t* infer_request) {
+    if (!infer_request) {
+        return ov_status_e::GENERAL_ERROR;
+    }
+
+    try {
+        infer_request->object->wait();
+    } CATCH_OV_EXCEPTIONS
+
+    return ov_status_e::OK;
+}
+
+ov_status_e ov_infer_request_set_callback(ov_infer_request_t* infer_request,
+                                        const ov_call_back_t* callback) {
+    if (!infer_request || !callback) {
+        return ov_status_e::GENERAL_ERROR;
+    }
+
+    try {
+        auto func = [=](std::exception_ptr ex) {
```

Comment:
I suggest to limit the capture list only to the required variables (just the callback in this case).